### ISSUE TITLE
Add well-known registry\file location to determine the global dotnet root location

### ISSUE THREAD

**Issue by steveharter** [dotnet/core-setup#4324]
This is a placeholder for the host portion of the [design proposal](https://github.com/dotnet/designs-microsoft/blob/bc49554ab2b5532556f541d4a9bb05fd7bc5e477/proposed/runtime/host-runtime-binding.md). Once this is approved, a new issue will be created for the installer portion.

The basic requirement is to probe a well-known registry location (Windows) or a well-known file (Unix) to determine the global location. The well-known locations:
- On Windows: `HKEY_LOCAL_MACHINE\SOFTWARE\dotnet\root` and `HKEY_LOCAL_MACHINE\SOFTWARE\dotnet\root\root(x86)`. (REG_SZ strings)
- On Unix: `/etc/dotnet/root`

The global location is to contain a single folder path, such as `C:\Program Files\dotnet`

For compat, if the registry key is not present on Windows, then the default global location will be used (`%ProgramFiles%\dotnet`  or `%ProgramFiles(x86)%\dotnet`). Unix did not have a global location, so there is no fallback. 

This global location will be used for the multi-level lookup feature which means the global location will always be used to find the best available runtime or SDK even if the dotnet.exe executable or apphost executable is executed in a different location. Unix did not have multi-level lookup support before since it did not have a global default location. The multi-level functionality can be disabled via %DOTNET_MULTILEVEL_LOOKUP% = 0.

Currently there is no plan to support probing %PATH% to find the global location as previously suggested (at least until discussion is closed).

The existing `%DOTNET_ROOT%` will continue to be used, and will processed before the global location (like it is today).

We need to discuss any potential porting to 2.1.x or 2.2, and potential breaking issues with multi-level being added to Unix in a non-major release.

**Comment by jkotas** [comment]
> `HKEY_LOCAL_MACHINE\SOFTWARE\dotnet\root\root(x86)` (REG_SZ strings)

Should this rather be `HKEY_LOCAL_MACHINE\SOFTWARE\dotnet\root` in the WoW registry hive?

**Comment by steveharter** [comment]
@jkotas the Wow hive sounds better. Updating that portion. Thanks

**Comment by johnbeisner** [comment]
Related: https://github.com/dotnet/core-setup/issues/4637
